packages/modular-scripts/src/__tests__/index.test.ts

```diff
@@ -421,16 +421,134 @@ describe('modular-scripts', () => {
         `https://mycustomcdn.net/lodash.merge?version=`,
       );
     });
+
+    it('THEN expects the correct bundledDependencies in package.json', async () => {
+      expect(
+        (
+          (await fs.readJson(
+            path.join(modularRoot, 'dist', 'sample-view', 'package.json'),
+          )) as CoreProperties
+        ).bundledDependencies,
+      ).toEqual(['lodash', 'lodash.merge']);
+    });
   });
 
-  it('THEN expects the correct bundledDependencies in package.json', async () => {
-    expect(
-      (
-        (await fs.readJson(
-          path.join(modularRoot, 'dist', 'sample-view', 'package.json'),
-        )) as CoreProperties
-      ).bundledDependencies,
-    ).toEqual(['lodash', 'lodash.merge']);
+  describe('WHEN building a view specifying a dependency to not being rewritten using wildcards', () => {
+    beforeAll(async () => {
+      await modular('build sample-view', {
+        stdio: 'inherit',
+        env: {
+          USE_MODULAR_ESBUILD: 'true',
+          EXTERNAL_CDN_TEMPLATE:
+            'https://mycustomcdn.net/[name]?version=[version]',
+          EXTERNAL_BLOCK_LIST: 'lodash*',
+        },
+      });
+    });
+
+    it('THEN outputs the correct directory structure', () => {
+      expect(tree(path.join(modularRoot, 'dist', 'sample-view')))
+        .toMatchInlineSnapshot(`
+        "sample-view
+        ├─ index.html #16orq8u
+        ├─ package.json
+        └─ static
+           └─ js
+              ├─ _trampoline.js #9qjmtx
+              ├─ index-P6RWJ53F.js #1emvouy
+              └─ index-P6RWJ53F.js.map #1y2yxmy"
+      `);
+    });
+
+    it('THEN rewrites only the dependencies that are not specified in the blocklist', async () => {
+      const baseDir = path.join(
+        modularRoot,
+        'dist',
+        'sample-view',
+        'static',
+        'js',
+      );
+
+      const indexFile = (
```

Comment:
I would just snapshot this so that we can assert the import syntax renames the same across esbuild upgrades

Reply:
Done, snapshotting all the index files.

Reply:
Probably best to use the format we have elsewhere where the file is run through prettier so that the snapshot is readable. See [here](https://github.com/jpmorganchase/modular/blob/main/packages/modular-scripts/src/__tests__/app.node-env.test.ts#L136) for an example from one of the other build tests

Reply:
Good catch, thanks. Done!